src/future/delay.rs

```diff
@@ -0,0 +1,61 @@
+use std::pin::Pin;
+use std::time::Duration;
+
+use futures_timer::Delay;
+
+use crate::future::Future;
+use crate::task::{Context, Poll};
+
+/// Creates a future that is delayed before it starts yielding items.
+///
+/// # Examples
+///
+/// ```
+/// # async_std::task::block_on(async {
+/// use async_std::future;
+/// use std::time::Duration;
+
+/// let a = future::delay(future::ready(1) ,Duration::from_millis(2000));
```

Comment:
I was thinking we make this method part of FutureExt (similar to how we define new stream methods) rather than expose it as a free function

Reply:
Like this?   b251fc9

Reply:
yeah, that's closer!